### PR TITLE
Implement pushing behavior on CognivitivePhysicalPedestrians

### PR DESCRIPTION
**Pull request by kelvin-olaiya**

### CODE REVIEW

File: alchemist-physical-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/nodes/CognitivePhysicalPedestrian2D.kt [hunk 1]
```diff
@@ -28,13 +29,30 @@ open class CognitivePhysicalPedestrian2D<T> @JvmOverloads constructor(
     group: PedestrianGroup2D<T>? = null
 ) : CognitivePedestrian2D<T>(environment, randomGenerator, age, gender, danger, group), PhysicalPedestrian2D<T> {
 
-    /**
-     * TODO(maybe introduce some randomness? To model differences in people)
+    /*
+        According to [the work of Pelechano et al](https://bit.ly/3e3C7Tb) in order to bring out
+        pushing behavior different nodes must have different personal space threshold.
```
Comment (DanySK) [inline alchemist-physical-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/nodes/CognitivePhysicalPedestrian2D.kt:34]: Use the same multiline comment we use elsewhere:
```kotlin
/*
 * text goes here after the '*'
 */
```

**Review by DanySK — CHANGES_REQUESTED** [review]
I am a bit puzzled by the fact that there are 42 commits, but very marginal changes and no additional tests. Is this a work in progress?
Since changes are pointwise, it would be fine to me to squash this in the mainline quickly, and then get on with the work with a simplified history.

**Comment by kelvin-olaiya** [comment]
Many of those commits were from previous pull requests. The ones regarding this PR are the last 14.

Sorry but i don't quite understand what you mean by squashing in the mainline and simplifing the history.

**Comment by DanySK** [comment]
Squashing means making all those commits a single commit. There are three techniques for unifying diverged branches:
* merging (creates a new commit, preserves the development history)
* squashing (compacts all commits of a pull request into a single commit)
* rebasing (preserves all commits, but re-arranges them in such a way that they build a linear history)

If I squash, I would recommend you to "start fresh" and continue development from the `develop` branch of the truth repository (this one).

**Review by DanySK — APPROVED** [review]

**Comment by DanySK** [comment]
I think squashing is the way to go here. If you wanted to summarize this PR with a single commit message, what'd you use?

**Comment by kelvin-olaiya** [comment]
Alright, I didn't realize that squashing is a git operation.

**Pull request merged**